### PR TITLE
ThreadPool: prefer std::mutex and std::condition_variable

### PR DESCRIPTION
**Pull request by rodolforg**
deprecated Glib::Threads::Mutex and Cond

### CODE REVIEW

File: synfig-core/src/synfig/threadpool.cpp [hunk 1]
```diff
@@ -241,25 +243,25 @@ ThreadPool::wakeup() {
...
 
 void
-ThreadPool::wait(Glib::Threads::Cond &cond, Glib::Threads::Mutex &mutex) {
+ThreadPool::wait(std::condition_variable &cond, std::mutex &mutex) {
 	if (--running_threads < max_running_threads)
 		if (queue_size) // wakeup or create ready thread if we have tasks in queue
-			{ Glib::Threads::Mutex::Lock lock(this->mutex); wakeup(); }
-	cond.wait(mutex);
+			{ std::lock_guard<std::mutex> lock(this->mutex); wakeup(); }
+	std::unique_lock<std::mutex> lock(mutex);
```
Comment (rodolforg) [inline synfig-core/src/synfig/threadpool.cpp]: I'm unsure about this...

File: synfig-core/src/synfig/threadpool.cpp [hunk 2]
```diff
@@ -117,7 +119,7 @@ ThreadPool::Group::run(bool force_thread) {
 
 	// wait
 	if (multithreading) {
-		Glib::Threads::Mutex::Lock lock(mutex);
+//		std::lock_guard<std::mutex> lock(mutex);
```
Comment (rodolforg) [inline synfig-core/src/synfig/threadpool.cpp]: I'm unsure about it.
wait() method locks it again (inside an `if` statement)

**Comment by ice0** [comment]
@rodolforg can you rebase it,  please? @blackwarthog made some changes to the original code.

**Comment by rodolforg** [comment]
@ice0  done :)

**Comment by ice0** [comment]
Merged. Thank you!

**Comment by rodolforg** [comment]
@ice0  I think it's hanging synfig on quit on master branch :(

**Comment by ice0** [comment]
@rodolforg I can't reproduce. How often this happens? Is it hangs when just start and quit, or I need to do some actions?

**Comment by rodolforg** [comment]
I can't investigate right now, but sometimes after rendering it halts too.
(I render via ffmpeg target, set to H.264, 1920x1080 or 1280x720).

**Pull request merged**